packages/vm/src/imports.rs

```diff
@@ -253,45 +234,19 @@ pub fn do_scan<S: Storage + 'static, Q: Querier>(
     start_ptr: u32,
     end_ptr: u32,
     order: i32,
-) -> i32 {
-    let start = match maybe_read_region(ctx, start_ptr, MAX_LENGTH_DB_KEY) {
-        Ok(data) => data,
-        Err(VmError::RegionLengthTooBigErr { .. }) => return errors::REGION_READ_LENGTH_TOO_BIG,
-        Err(_) => return errors::REGION_READ_UNKNOWN,
-    };
-    let end = match maybe_read_region(ctx, end_ptr, MAX_LENGTH_DB_KEY) {
-        Ok(data) => data,
-        Err(VmError::RegionLengthTooBigErr { .. }) => return errors::REGION_READ_LENGTH_TOO_BIG,
-        Err(_) => return errors::REGION_READ_UNKNOWN,
-    };
+) -> VmResult<i32> {
+    let start = maybe_read_region!(ctx, start_ptr, MAX_LENGTH_DB_KEY);
+    let end = maybe_read_region!(ctx, end_ptr, MAX_LENGTH_DB_KEY);
     let order: Order = match order.try_into() {
-        Ok(o) => o,
-        Err(_) => return errors::scan::INVALID_ORDER,
+        Ok(order) => order,
```

Comment:
I hate how Rust encurages variable shadowing. But for some reason it is common. Anyways, I'll get used to it at some point or find a liter that blocks it :)

Reply:
I personally find it to be a great feature, especially in these circumstances where the `Ok` part is the part we really care about, and the naming reflects it, but i agree that it can be confusing to people that aren't used to it, and can be seen as ugly, even. If you want to avoid it, i think it's most appropriate to rename the `order` in the `order.try_into()` expression.

Reply:
The code is good as-is. I'm just always surprised that super strict and safe Rust allows that, even without a clippy rule. So many things can go wrong with shadow variables during refactoring.

Reply:
actually yeah, variable renaming is dangerous around these...